### DISCUSSION TITLE
Using Rocket as a CGI Handler?

### DISCUSSION THREAD

**Discussion by dpavlic** [rwf2/Rocket#2370] [Questions]
I see there is some discussion in GitHub issues about using Rocket to handle CGI, but all of the discussion on this seems from quite a while ago (2017). Is this actually possible to do with Rocket?

**Comment by antimeme** [comment]
I'm interested in this as well.  I don't yet know enough to be able to solve it, but I suspect it might start with std::process::Child (https://doc.rust-lang.org/std/process/struct.Child.html).  This would have to be wrapped with a custom Responder from Rocket.  I don't see any mention of async or await in the Child documentation, so maybe this requires spawning a worker.  Error handling is sure to be... interesting.

**Reply by the10thWiz** [reply]
`std::process::Child` doesn't support async. You would be looking for `tokio::process::Child`, which is the appropriate async implementation to use with Rocket.

**Reply by antimeme** [reply]
That makes sense.  Thanks.

**Comment by antimeme** [comment]
This might also be part of the solution:

https://github.com/SergioBenitez/Rocket/discussions/2368

I imagine waiting on Child might be done this way.

**Comment by the10thWiz** [comment]
For anyone still interested: I've written a CGI library to match the CGI RFC. It's still a WIP right now, but I've published a version of it to Crates.io. https://github.com/the10thWiz/rocket-cgi

@antimeme @dpavlic
